index.js

```diff
@@ -9,11 +9,10 @@ function blind (priv) {
 
 function getr (priv) {
   var len = priv.modulus.byteLength()
-  var r = new BN(randomBytes(len))
-  while (r.cmp(priv.modulus) >= 0 || !r.umod(priv.prime1) || !r.umod(priv.prime2)) {
-    r = new BN(randomBytes(len))
+  while (true) {
+    var r = new BN(randomBytes(len))
+    if (r.cmp(priv.modulus) < 0 && r.umod(priv.prime1) && r.umod(priv.prime2)) return r
   }
-  return r
```

Comment:
This code,  but leave `r` uninitialized, and `s/while/do` and `s/}/} while (...)`